### PR TITLE
test(CL): swap to new min tick and back

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #https://github.com/osmosis-labs/osmosis/issues/6061

## What is the purpose of the change

This test validates that swapping over the new extended price range in the lower
direction functions as expected. It validates that there are no unexpected failures
or panics when swapping to the new min tick and back.
Additionally, it validates that the swap amounts are roughly equal to the inverse amounts of a given swap.

Adds helpers for setting up positions and basic state needing to test this.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [x] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by p0mvn** [comment]
Please do not merge until https://github.com/osmosis-labs/osmosis/pull/6323 is in

File: x/concentrated-liquidity/keeper_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/keeper_test.go:481]: Note to reviewer: drive by change that is not needed anymore

File: x/concentrated-liquidity/swaps_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/swaps_test.go:3593]: Note to self: update comment before merge

File: x/concentrated-liquidity/swaps_test.go
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/swaps_test.go:3528]: Do you think it would be appropriate to also have a version that swaps to `MinCurrentTick`? Is this even possible?

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, nice work

File: x/concentrated-liquidity/swaps_test.go
Comment (p0mvn) [reply x/concentrated-liquidity/swaps_test.go:3593]: I'm running into issues with this test case in: https://github.com/osmosis-labs/osmosis/pull/6341

and debugging it there. I will merge this as is right now with the promise to address this in #6341

File: x/concentrated-liquidity/swaps_test.go
Comment (p0mvn) [reply x/concentrated-liquidity/swaps_test.go:3528]: It should reach the `MinCurrentTick` if all liquidity is consumed in the full range and we cross `types.MinInitializedTick`.

This is expected to happen hear. Added assertions to make this clear:
[a2dbf41](https://github.com/osmosis-labs/osmosis/pull/6327/commits/a2dbf41fc94227def810879ba22f7a7eca40182d)

File: x/concentrated-liquidity/swaps_test.go
Comment (p0mvn) [reply x/concentrated-liquidity/swaps_test.go]: addressed: [6ddb612](https://github.com/osmosis-labs/osmosis/pull/6327/commits/6ddb61211c655254568d44e9d2953c786cda816a)

**Pull request merged**
